sdk/request_utils.go

```diff
@@ -0,0 +1,92 @@
+package sdk
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"net/http"
+	"net/url"
+	"strconv"
+)
+
+func handlePotentialManyRequestsResponse(httpResponse *http.Response, error error) {
+	if error == nil {
+		return
+	}
+
+	var e GenericSwaggerError
+
+	if errors.As(error, &e) {
+		if model, ok := e.model.(*ManyRequestsResponse); ok {
+			retryAfter := getRetryAfterFromHeader(httpResponse)
+
+			model.RetryAfter = retryAfter
+		}
+	}
+}
+
+func getRetryAfterFromHeader(httpResponse *http.Response) int64 {
+	header := httpResponse.Header.Get("Retry-After")
+
+	if header != "" {
+		parsed, err := strconv.ParseInt(header, 10, 32)
+
+		if err == nil {
+			return parsed
+		}
+	}
+
+	return 0
+}
+
+func getQueryWithIntegrationInfo(url *url.URL) url.Values {
+	query := url.Query()
+
+	query.Add("ii", IntegrationInfo)
+
+	return query
+}
+
+func handleErrorResponse[T any](body []byte, httpResponse *http.Response, definition requestDefinition, responseModel T) (T, *http.Response, error) {
+	genericError := GenericSwaggerError{
+		body:  body,
+		error: httpResponse.Status,
+	}
+
+	modelFactory := definition.StatusCodeResultsFactoryMap[httpResponse.StatusCode]
+
+	if modelFactory != nil {
+		model := modelFactory()
+
+		err := json.Unmarshal(body, &model)
+
+		if err != nil {
+			genericError.error = err.Error()
+
+			return responseModel, httpResponse, genericError
+		}
+
+		genericError.model = model
+	}
+
+	return responseModel, httpResponse, genericError
+}
+
+func isResponseOk(httpResponse *http.Response) bool {
+	return httpResponse.StatusCode < 300
```

Comment:
should be appended with `return 199 < httpResponse.StatusCode && ...`

Reply:
Does it make sense? I'm not sure if there is even a possibility that your API would send status code smaller than 200.

Reply:
It is still not the correct implementation. The definition of a successful (or Ok) HTTP request based on the response is determined by the status code being greater than 199 and less than 300.

Reply:
Fair enough, done!